### ISSUE TITLE
SiteConfigurationReader.mergeSiteFilesWithCache throws ConcurrentModificationException in Java 9

### ISSUE THREAD

**Issue by michaelhixson** [MTDdk/jawn#31]
I'm trying to run [jawn's implementation of the TechEmpower Framework Benchmarks](https://github.com/TechEmpower/FrameworkBenchmarks/tree/master/frameworks/Java/jawn) on Java 9 (as part of a broader effort to migrate from Java 8 to 9), and when I do this the fortunes test fails with an exception.  I think this is an issue with the framework rather than the benchmark implementation, which is why I'm posting this here.

Here's the stack trace that I see:

```
java.util.ConcurrentModificationException: null
   at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1139) ~[na:na]
   at net.javapla.jawn.core.templates.config.SiteConfigurationReader.mergeSiteFilesWithCache(SiteConfigurationReader.java:145) ~[jawn-core-0.9.10.jar:na]
   at net.javapla.jawn.core.templates.config.SiteConfigurationReader.read(SiteConfigurationReader.java:94) ~[jawn-core-0.9.10.jar:na]
   at net.javapla.jawn.core.templates.stringtemplate.StringTemplateTemplateEngine.injectValuesIntoLayoutTemplate(StringTemplateTemplateEngine.java:254) ~[jaw                             n-core-0.9.10.jar:na]
   at net.javapla.jawn.core.templates.stringtemplate.StringTemplateTemplateEngine.invoke(StringTemplateTemplateEngine.java:135) ~[jawn-core-0.9.10.jar:na]
   at net.javapla.jawn.core.ResultRunner.renderWithTemplateEngine(ResultRunner.java:60) ~[jawn-core-0.9.10.jar:na]
   at net.javapla.jawn.core.ResultRunner.run(ResultRunner.java:41) ~[jawn-core-0.9.10.jar:na]
   at net.javapla.jawn.core.FrameworkEngine.onRouteRequest(FrameworkEngine.java:70) ~[jawn-core-0.9.10.jar:na]
   at net.javapla.jawn.core.HttpHandlerImpl.handle(HttpHandlerImpl.java:52) [jawn-core-0.9.10.jar:na]
   at net.javapla.jawn.server.undertow.UndertowHandler.handleRequest(UndertowHandler.java:24) [jawn-server-undertow-0.9.10.jar:na]
   at io.undertow.server.Connectors.executeRootHandler(Connectors.java:332) [undertow-core-1.4.20.Final.jar:1.4.20.Final]
   at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:812) [undertow-core-1.4.20.Final.jar:1.4.20.Final]
   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) [na:na]
   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) [na:na]
   at java.base/java.lang.Thread.run(Thread.java:844) [na:na]
```

The throwing statement is this call to `computeIfAbsent`:

https://github.com/MTDdk/jawn/blob/a8994b1a9fb2bbfe00994d78caab4c7ce27acaaa/jawn-core/src/main/java/net/javapla/jawn/core/templates/config/SiteConfigurationReader.java#L144-L147

I think the issue is that the key remapping function (on line 146) modifies the `configurationCache` map (it calls `readSiteFileWithCache`, which modifies the map on line 135).  That's a problem because that map is a `HashMap`, and `HashMap` prohibits this kind of modification as of Java 9:

https://docs.oracle.com/javase/9/docs/api/java/util/HashMap.html#computeIfAbsent-K-java.util.function.Function-
> This method will, on a best-effort basis, throw a `ConcurrentModificationException` if it is detected that the mapping function modifies this map during computation.

I'm not sure what the best solution is.  Perhaps you could avoid `computeIfAbsent` in favor of checking and updating the map in separate statements the old-fashioned way, since it's a `HashMap` and not concurrent / thread-safe anyway.

**Comment by MTDdk** [comment]
Thanks a lot for the notice!

We'll take a look at it right away

**Comment by MTDdk** [comment]
fixed by 03971169ce038c43a27d58cedd6d91599a3557b0
